Updating the search relevance api

ajleong623 opened the pull request:
### Description
There are changes and new apis that should be represented in the specification. Additionally, I have to change what is making the smoke test fail and remove the Dockerfile since 3.1.0 is coming out.

### Issues Resolved
#919 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

martin-gaievski commented:
In addition to changes to existing APIs we need to onboard one more new API for stats, main purpose for it is to collect statistics on usage of plugin components.
this is sample request, it only take GET requests
```
curl  localhost:9200/_plugins/_search_relevance/stats
```
and sample response:
```
{
  "_nodes": {
    "total": 1,
    "successful": 1,
    "failed": 0
  },
  "cluster_name": "integTest",
  "info": {
    "cluster_version": "3.1.0"
  },
  "all_nodes": {
    "judgments": {
      "import_judgment_rating_generations": 0,
      "llm_judgment_rating_generations": 0,
      "ubi_judgment_rating_generations": 0
    },
    "experiments": {
      "experiment_hybrid_optimizer_executions": 0,
      "experiment_pairwise_comparison_executions": 0,
      "experiment_executions": 0,
      "experiment_pointwise_evaluation_executions": 0
    }
  },
  "nodes": {
    "v5oHyj8yTRSuUKrmf1UoIw": {
      "judgments": {
        "import_judgment_rating_generations": 0,
        "llm_judgment_rating_generations": 0,
        "ubi_judgment_rating_generations": 0
      },
      "experiments": {
        "experiment_hybrid_optimizer_executions": 0,
        "experiment_pairwise_comparison_executions": 0,
        "experiment_executions": 0,
        "experiment_pointwise_evaluation_executions": 0
      }
    }
  }
}
```

dblock commented on tests/plugins/search_relevance/Dockerfile:
```diff
@@ -1,4 +1,5 @@
 FROM opensearchstaging/opensearch:3.1.0
 
+RUN /usr/share/opensearch/bin/opensearch-plugin remove opensearch-search-relevance
```
This one is odd. The plugin is included in the distribution? Why does it need to be uninstalled?

ajleong623 commented:
So, for the updates, I added the general matrix entry as suggested in the other pull request, and I also removed the parts of the Dockerfile that use the temporary s3 bucket since the plugin is now available.

ajleong623 commented:
For some reason, the general 3.1.0 version is not passing the tests properly. I think that some apis might not be up to date because the neural/stats test paths are not passing. For now, I might have to leave it out.

dblock commented:
> For some reason, the general 3.1.0 version is not passing the tests properly. I think that some apis might not be up to date because the neural/stats test paths are not passing. For now, I might have to leave it out.

You should investigate this.

Xtansia commented:
> > For some reason, the general 3.1.0 version is not passing the tests properly. I think that some apis might not be up to date because the neural/stats test paths are not passing. For now, I might have to leave it out.
> 
> You should investigate this.

This is already being covered in #923
